sysid-application/src/main/native/cpp/analysis/AnalysisManager.cpp

```diff
@@ -625,6 +625,10 @@ void AnalysisManager::PrepareData() {
 }
 
 AnalysisManager::Gains AnalysisManager::Calculate() {
+  if (m_filteredDatasets.empty()) {
+    throw std::runtime_error("Invalid data to perform gain calculation.");
```

Comment:
What makes the data invalid and why did it happen? This error message isn't helpful.

Reply:
In essence, this will run if there was an error with the data analysis (e.g. the quasistatic trim) and the user manages to trigger the gain calculation function.

Would this work:
"There is an unresolved issue with the data being used. Please double-check the data quality and adjust settings such as units, velocity threshold, test duration, etc. to make sure that there are no errors with the dataset before attempting to calculate gains."

Reply:
Yea.